core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -983,17 +985,16 @@ protected R loadBuild(File dir) throws IOException {
      * @see TransientProjectActionFactory
      */
     @SuppressWarnings("deprecation")
+    @Nonnull
     @Override
     public List<Action> getActions() {
         // add all the transient actions, too
-        List<Action> actions = new Vector<Action>(super.getActions());
+        List<Action> actions = new Vector<>(super.getActions());
```

Comment:
Maybe

```suggestion
        List<Action> actions = new Vector<>(getPersistedActions());
```

for clarity.